# Changelog

## [0.10.0](https://www.github.com/hydra-genetics/cnv_sv/compare/v0.9.0...v0.10.0) (2025-08-27)


### Features

* add savana ([1805178](https://www.github.com/hydra-genetics/cnv_sv/commit/1805178628156c3cd137c4df14d8396e2e355191))


### Bug Fixes

* add dir entry to savana_* rules ([6ff1ec0](https://www.github.com/hydra-genetics/cnv_sv/commit/6ff1ec03dc7a9c8e93e9bcf9cadc54b8ad0fa102))
* add genome version to params in the TO rules ([6c6ef84](https://www.github.com/hydra-genetics/cnv_sv/commit/6c6ef847a0291e5a5dbaec5e7957938ca404af0b))
* add genome version to the TO rules ([cf5da60](https://www.github.com/hydra-genetics/cnv_sv/commit/cf5da608e233ff744c1402e6a0a8685a18cd96ec))
* add genome_version to the TO rules ([ce3c5b6](https://www.github.com/hydra-genetics/cnv_sv/commit/ce3c5b6545ed2929568dc6626011a018aa672aa8))
* add min_support to parameters ([c4f4af6](https://www.github.com/hydra-genetics/cnv_sv/commit/c4f4af62c77e9b343959af5eed9dc166cdd60a80))
* add output description to savana_* rules ([9dcf3d0](https://www.github.com/hydra-genetics/cnv_sv/commit/9dcf3d0a9225c59bb731ad0e59de36644fba9ac2))
* add savana_* entries to config.yaml ([98f75e0](https://www.github.com/hydra-genetics/cnv_sv/commit/98f75e09332a07d83749e7d12ae98339fc0e2374))
* alphabetical order of the savana_* rules in rules.schema.yaml ([238f9ce](https://www.github.com/hydra-genetics/cnv_sv/commit/238f9ce6fcb564aeb533e75fd3892488a4f38b25))
* full output filenames & outdir to params ([5d23ac2](https://www.github.com/hydra-genetics/cnv_sv/commit/5d23ac28507a1d1a7f7a8e1dba1eeb2518ec397c))


### Documentation

* add description of savana_* rules to softwares.md ([a2d88c7](https://www.github.com/hydra-genetics/cnv_sv/commit/a2d88c790e59be2773ad173b0bf32ea0af7a28c5))
* add entries for savana_* rules to config.schema.yaml ([8a14af8](https://www.github.com/hydra-genetics/cnv_sv/commit/8a14af8fcc9b80837b12131e735ce89127c003de))
* add schemas for rules savana_* ([c9b1a43](https://www.github.com/hydra-genetics/cnv_sv/commit/c9b1a43b9df5ca13a3b542f08d4980605424039e))
* update DAGs and README.md ([47afb28](https://www.github.com/hydra-genetics/cnv_sv/commit/47afb28ec3545f1ab5d1f4e0460d440986c6f602))
* update docs/softwares.md ([d5fdce8](https://www.github.com/hydra-genetics/cnv_sv/commit/d5fdce851f04adb90f60f8588b387e99be8024bb))
* update docs/softwares.md ([688c081](https://www.github.com/hydra-genetics/cnv_sv/commit/688c08145e10f4cbed7c608cf346dc52353ec07b))
* update docs/softwares.md ([d76e841](https://www.github.com/hydra-genetics/cnv_sv/commit/d76e8411c74cec9f73e350bd48d9eb3254050720))
* update docs/softwares.md ([4268631](https://www.github.com/hydra-genetics/cnv_sv/commit/4268631f82c8fef281c584ccc25a9949f22f4ade))
* update the module output files ([02aca91](https://www.github.com/hydra-genetics/cnv_sv/commit/02aca91417bfabed6eb6ee1dc8583c609b9cc02b))

## [0.9.0](https://www.github.com/hydra-genetics/cnv_sv/compare/v0.8.0...v0.9.0) (2025-05-14)


### Features

* added scanITD ([6b7e9bf](https://www.github.com/hydra-genetics/cnv_sv/commit/6b7e9bf5a6ecf48c518f4542f9a6bfb2db383661))


### Bug Fixes

* optional bed file ([7e2d882](https://www.github.com/hydra-genetics/cnv_sv/commit/7e2d8821e7206859d58532f0889455f2abf01ef3))
* **scanITD:** correct rule params ([01cb157](https://www.github.com/hydra-genetics/cnv_sv/commit/01cb157b762dd03023eea58aefcadd74a2fde9a6))
* **scanITD:** correct shell command ([3da1a76](https://www.github.com/hydra-genetics/cnv_sv/commit/3da1a76bf95766149f9d750f9472d8ec3ed1b185))
* update to bugfixed docker ([20477be](https://www.github.com/hydra-genetics/cnv_sv/commit/20477be5d70c92803b258398089e478015289f6b))


### Documentation

* added docs and tests ([0fc6464](https://www.github.com/hydra-genetics/cnv_sv/commit/0fc64646e1049d93a85021ce1a890220d37582f9))

## [0.8.0](https://www.github.com/hydra-genetics/cnv_sv/compare/v0.7.1...v0.8.0) (2025-05-05)


### Features

* add entries for severus_t_only and severus_tn ([89ec119](https://www.github.com/hydra-genetics/cnv_sv/commit/89ec119182c830a3280a5db585abd80468ebc883))
* add function to compile input BAM paths ([49c2f1c](https://www.github.com/hydra-genetics/cnv_sv/commit/49c2f1c15452863db3fa9990a1cff8f7d98491ca))
* add paraphase ([2d533ab](https://www.github.com/hydra-genetics/cnv_sv/commit/2d533ab7eda83cc077803e9751f33ccae60ac05c))
* add pbsv.smk ([3854216](https://www.github.com/hydra-genetics/cnv_sv/commit/385421622d3a7a7590b9b33fbc80515a55f3e322))
* add rule for HiFiCNV ([930b800](https://www.github.com/hydra-genetics/cnv_sv/commit/930b8008ea60cc36a3d9fe45d34018909aa08442))
* add sawfish ([e7408b8](https://www.github.com/hydra-genetics/cnv_sv/commit/e7408b8d30c29613e389123c5fe4c87dbfa9637a))
* add severus related entries to config.yaml ([cd64c98](https://www.github.com/hydra-genetics/cnv_sv/commit/cd64c9802e938117f41e2233a31a363164611f31))
* add severus.smk ([53c64dc](https://www.github.com/hydra-genetics/cnv_sv/commit/53c64dc0924901482bd85cb1f6f3b39dbc6b9137))
* add severus.smk to the list of included rules ([39a0dd0](https://www.github.com/hydra-genetics/cnv_sv/commit/39a0dd0208f1f9bf33b3b99a899880e9a091e869))
* include hificnv.smk and pbsv.smk ([76ed3fc](https://www.github.com/hydra-genetics/cnv_sv/commit/76ed3fc59f46b5cc661b97c06f1e67300bcd2203))
* new name and functionality of the functon for creating input bam files ([2b21dec](https://www.github.com/hydra-genetics/cnv_sv/commit/2b21dece2f6c8c38d3d336258c4a70793327e243))
* update severus version to 1.5 ([e227d4f](https://www.github.com/hydra-genetics/cnv_sv/commit/e227d4f37155ecf6f7e3fb09da472b3b92f16659))
* use get_input_bam() in rule cnvkit_batch ([db571ba](https://www.github.com/hydra-genetics/cnv_sv/commit/db571ba3ed66f13b057e0abf575dd2afd89e8101))


### Bug Fixes

* add 'exclude' entry and 'exclude' file for hificnv ([f9c9bf5](https://www.github.com/hydra-genetics/cnv_sv/commit/f9c9bf50cba19fc70a08bf4ae5c39e4a7d452a45))
* add {type} to input, output, log and benchmark files in pbsv_call ([6c5c78c](https://www.github.com/hydra-genetics/cnv_sv/commit/6c5c78cf1b7417988114d65e8b24db7e9dad6944))
* add {type} to ouput, log and benchmark files in pbsv_discover ([f4ca49c](https://www.github.com/hydra-genetics/cnv_sv/commit/f4ca49c69802b0eb7656e1ffa90eab66eba417de))
* add {type} to output, log and benchmark files in rule hificnv ([26b5c38](https://www.github.com/hydra-genetics/cnv_sv/commit/26b5c38a4aac977af85a3b335389fabb95d75abf))
* add {unit_type} to output files of pbsv_call, pbsv_discover and hificnv in compile_output_list() ([71b2572](https://www.github.com/hydra-genetics/cnv_sv/commit/71b2572f902ae8db0a54cb87b5eb1b606dcb825d))
* add all output files description to severus_* ([8705dce](https://www.github.com/hydra-genetics/cnv_sv/commit/8705dce34522631f47433085744f25be922e8428))
* add all required sub-entries to severus entries ([ee36340](https://www.github.com/hydra-genetics/cnv_sv/commit/ee36340ef96d5fcce60590c001824e86b17d82d8))
* add container for hificnv to .tests/config_pacbio.yaml ([c1fa21a](https://www.github.com/hydra-genetics/cnv_sv/commit/c1fa21a1b4633dc4970921fb474954a649d7ec47))
* add docker image path, reference and trf files to pbsv rules ([4842a93](https://www.github.com/hydra-genetics/cnv_sv/commit/4842a93e128db7f723ebea5a0138eceea81a198f))
* add entries for hificnv and pbsv ([624d28b](https://www.github.com/hydra-genetics/cnv_sv/commit/624d28b16292af848c290d85364c8d4b54e649d2))
* add output files from hificnv and pbsv to compile_output_list() ([d7241dd](https://www.github.com/hydra-genetics/cnv_sv/commit/d7241dd6c3a2199b07d561504e97d4f2f89bba87))
* add renaming of the output files after HiFiCNV run ([762fbd7](https://www.github.com/hydra-genetics/cnv_sv/commit/762fbd79d0f9ecc74377c585cac505e657d97d62))
* add required quotation marks in awk command ([ccdc1fd](https://www.github.com/hydra-genetics/cnv_sv/commit/ccdc1fd373c3bb5ac3b709b0e73582d61f3f366d))
* add temp() directive to output ([0c4add0](https://www.github.com/hydra-genetics/cnv_sv/commit/0c4add03f30a1322a1efdc41a29e3dc585aa9774))
* change input BAM files in pbsv_discover from hard-coded to function ([cab49eb](https://www.github.com/hydra-genetics/cnv_sv/commit/cab49ebb15aecac2e6b53d62b4af7b333b54f00c))
* change input bam path; mv ref to params ([6075a12](https://www.github.com/hydra-genetics/cnv_sv/commit/6075a12fdfd8d355c1ee391b8168616ec9d9fe7f))
* change input from  path to function ([8b7c139](https://www.github.com/hydra-genetics/cnv_sv/commit/8b7c139bff70774c2953654ba559c72d7a16bb6f))
* change output/log to vcf.gz ([7d409db](https://www.github.com/hydra-genetics/cnv_sv/commit/7d409dbf66e7e74cf6204e8e3a14afa9623dd47a))
* correct alignment_path and index_path in get_input_bam() ([d232314](https://www.github.com/hydra-genetics/cnv_sv/commit/d2323143815b5bc19854817f8bbd4fdb4a539b38))
* correct config entry name for pon ([c867311](https://www.github.com/hydra-genetics/cnv_sv/commit/c867311cd4e644b78ea21c10386bb5010ea3f25b))
* correct output subdir name for severus_tn ([74c0648](https://www.github.com/hydra-genetics/cnv_sv/commit/74c064801b0ab0efcb7c1a7b10e476852a36aeb0))
* Delete workflow/rules/juli.smk ([d72fc1e](https://www.github.com/hydra-genetics/cnv_sv/commit/d72fc1e5dd68333144439d86cb24fe8c4abc1abe))
* limit number of threads used by Jumble to that specified in config ([b27fb17](https://www.github.com/hydra-genetics/cnv_sv/commit/b27fb174baa146d5492e787232f6cb4a3805477f))
* make two entries for pbsv (call & discover) ([c8a9272](https://www.github.com/hydra-genetics/cnv_sv/commit/c8a9272e129f1bd49c9b355ada28353dc075995f))
* make two entries for pbsv (call & discover) ([55b8793](https://www.github.com/hydra-genetics/cnv_sv/commit/55b8793b11a3b0688a42beea99474c458251bdfa))
* **paraphase:** add missing extra to shell command ([1250f24](https://www.github.com/hydra-genetics/cnv_sv/commit/1250f243de2623e1c235a26258ac32febceafa45))
* remove print() from get_input_bam() ([80800a4](https://www.github.com/hydra-genetics/cnv_sv/commit/80800a4e1724d0920f6a6ea7a1487f2ec3fa57f7))
* **sawfish:** update inputs to function ([671b316](https://www.github.com/hydra-genetics/cnv_sv/commit/671b316e8f2c7410c08447f0ae959e3862bbd9c0))
* separate output files not only dir ([f977330](https://www.github.com/hydra-genetics/cnv_sv/commit/f97733055ef615689608e95680864fa528b69058))
* **sniffles2:** add sample-id param to add sample name in VCF ([7f236b9](https://www.github.com/hydra-genetics/cnv_sv/commit/7f236b990664d66de412663bc4cf4913836187bf))
* specify all output files and fix the 'ouput file name as prefix' problem ([8d9490e](https://www.github.com/hydra-genetics/cnv_sv/commit/8d9490ef4c232a1c0dcf8124b22f0af79a612ae0))
* **tiddit:** add threads and access extras ([330c506](https://www.github.com/hydra-genetics/cnv_sv/commit/330c5064b76edbb8205b6d630f311a3dae3ff4ca))
* use correct name of the input bam function ([0ff7ccd](https://www.github.com/hydra-genetics/cnv_sv/commit/0ff7ccdf1c4eb70f3990fb25bf960c848e2c7de6))
* use get_input_bam() ([5aeb9af](https://www.github.com/hydra-genetics/cnv_sv/commit/5aeb9af6deb282743a7191430f78bd8b63ccf85a))


### Documentation

* update schemas with sawfish info ([e8f0163](https://www.github.com/hydra-genetics/cnv_sv/commit/e8f0163b632499f4f81908dc5d76d7d24c48e763))

### [0.7.1](https://www.github.com/hydra-genetics/cnv_sv/compare/v0.7.0...v0.7.1) (2024-10-16)


### Bug Fixes

* add missing parenthesis ([9328c59](https://www.github.com/hydra-genetics/cnv_sv/commit/9328c593d2bc0b3b525ef1fd8214b90e9ed2ee4a))
* jumble results removed prematurely ([5ae3b74](https://www.github.com/hydra-genetics/cnv_sv/commit/5ae3b74b4da103c03332c77f2caa19aed5e2bee1))
* **tiddit:** add bai file as an input ([ace4bcd](https://www.github.com/hydra-genetics/cnv_sv/commit/ace4bcd17b2f7c16f58b9eac6b223308067a9d45))


### Documentation

* update documentation ([087f67e](https://www.github.com/hydra-genetics/cnv_sv/commit/087f67e52ce4cda10861f58f4c42609b354dad95))
* update mkdocs ([e28d589](https://www.github.com/hydra-genetics/cnv_sv/commit/e28d5896c9dbd3727bcff704bf6e7967b2b43bf0))

## [0.7.0](https://www.github.com/hydra-genetics/cnv_sv/compare/v0.6.0...v0.7.0) (2024-10-02)


### Features

* add possibility to change caller annotation ([267285f](https://www.github.com/hydra-genetics/cnv_sv/commit/267285f782855a7903c4165ef829c515d2082aa2))
* add possibility to change caller name ([0b5f000](https://www.github.com/hydra-genetics/cnv_sv/commit/0b5f000f761b934d4ffc747b432f00efdffdebec))
* make caller name configurable ([461aee1](https://www.github.com/hydra-genetics/cnv_sv/commit/461aee1d4d14bc2cf337c8c634eebe2148827325))


### Bug Fixes

* make name configurable ([69ea10a](https://www.github.com/hydra-genetics/cnv_sv/commit/69ea10ac806fd0fec770a72e3974d7e1b6a18a14))
* put param in correct rule ([b99712e](https://www.github.com/hydra-genetics/cnv_sv/commit/b99712e44513dc7840a1da9fac5b0244ecb60b15))
* set sample_name correctly in header ([ef428c6](https://www.github.com/hydra-genetics/cnv_sv/commit/ef428c6d69394fd8495eaf8ebfe5db730932f475))

## [0.6.0](https://www.github.com/hydra-genetics/cnv_sv/compare/v0.5.0...v0.6.0) (2024-09-12)


### Features

* add chr to chromosome if missing ([0197bb2](https://www.github.com/hydra-genetics/cnv_sv/commit/0197bb2c48b9d45d72c9d562a18ac02169511508))
* add sniffles2 single sample calling ([0ce9b8a](https://www.github.com/hydra-genetics/cnv_sv/commit/0ce9b8aa3821e4c98d78d8f89ca8d423164e1765))
* add tabix and bgzip for vcf output ([f29e4a2](https://www.github.com/hydra-genetics/cnv_sv/commit/f29e4a29c469b931137f64a9698ccdb3c786c744))
* add trgt ([145641d](https://www.github.com/hydra-genetics/cnv_sv/commit/145641daefdd6e47d9dc07b229c964582f8874d6))
* added cnvkit call of jumble output ([605ce90](https://www.github.com/hydra-genetics/cnv_sv/commit/605ce9040f4137aead0a1f392d34cc0d931ee5fe))
* added jumble_vcf rule for segment exports ([0aa2632](https://www.github.com/hydra-genetics/cnv_sv/commit/0aa263211e11c6f16c56b1adb47da832cb87b905))
* added the CNV caller Jumble ([4d24aa8](https://www.github.com/hydra-genetics/cnv_sv/commit/4d24aa883305ac8783af0df91f9b9e02e2071545))


### Bug Fixes

* add default values for pacbio_alignment and ont_alignment in config.schema.yaml ([ddedade](https://www.github.com/hydra-genetics/cnv_sv/commit/ddedade1728750d9cdd64f18a4d53532a05457ac))
* change to compatible wrapper ([4759cff](https://www.github.com/hydra-genetics/cnv_sv/commit/4759cffbb9cbd2ac24efb85296cb9a854fcc19ef))
* code improvement ([911422a](https://www.github.com/hydra-genetics/cnv_sv/commit/911422a2cd2428c574b7f398c8129f0a340577a6))
* **common:** set type to str for pandas dataframe ([4c0e0f2](https://www.github.com/hydra-genetics/cnv_sv/commit/4c0e0f2252da1ab187ff94c132523fe1958c959c))
* correct input parameter name ([1722db4](https://www.github.com/hydra-genetics/cnv_sv/commit/1722db4fea07392d23eb31c13bf6bd4586fb5ae6))
* corrected output file names ([238fb36](https://www.github.com/hydra-genetics/cnv_sv/commit/238fb36ac6e9ffe4bb6318ef244e564e5f7719ce))
* jumble_vcf should have same parameters as cnvkit_vcf ([1748415](https://www.github.com/hydra-genetics/cnv_sv/commit/1748415286965c99ea7864c5fc11694f3f8f6568))
* make sure sample column is a string ([2d3aa6b](https://www.github.com/hydra-genetics/cnv_sv/commit/2d3aa6bae5addca7219e9aa833778110efd1173f))
* missing parenthesis ([be912f8](https://www.github.com/hydra-genetics/cnv_sv/commit/be912f822c7f7284676c7b083dc51214da5bb9d8))
* update common container ([409dbda](https://www.github.com/hydra-genetics/cnv_sv/commit/409dbda6607a2d6cbba60ea63632bebb9893e333))
* Update config_pacbio.yaml ([e7f6117](https://www.github.com/hydra-genetics/cnv_sv/commit/e7f6117eac6d519f1238395b0bb79864b6d9daa9))
* Update config.yaml ([c2a82f8](https://www.github.com/hydra-genetics/cnv_sv/commit/c2a82f81e283dabec54a7e8494ba1adf41a2b574))
* Update config.yaml ([ea9a0f7](https://www.github.com/hydra-genetics/cnv_sv/commit/ea9a0f7d916952197381fcb61c98063a2f71835b))
* Update config.yaml ([794e3d6](https://www.github.com/hydra-genetics/cnv_sv/commit/794e3d6fedc8d26674128bb7f06db814e7900d44))
* wrong config parameter name ([3ad5336](https://www.github.com/hydra-genetics/cnv_sv/commit/3ad533667c4ea0577793c7e55cdb3cf3474cb23e))


### Documentation

* documentation fix ([958550b](https://www.github.com/hydra-genetics/cnv_sv/commit/958550be4c0b91ac499f73310ce4f4078e76b6b0))
* update rules ([e1166af](https://www.github.com/hydra-genetics/cnv_sv/commit/e1166aff1bee8ace4b6f4a8583cfd9d7a4cd7218))

## [0.5.0](https://www.github.com/hydra-genetics/cnv_sv/compare/v0.4.1...v0.5.0) (2024-03-28)


### Features

* add names for the vcfs using params and add bnd_distance param ([e32ca47](https://www.github.com/hydra-genetics/cnv_sv/commit/e32ca47bfbac69ef37bf670c9a014937c0cfd037))
* add priority as a param ([db6e1ba](https://www.github.com/hydra-genetics/cnv_sv/commit/db6e1bac37321f23a7d2763b94493ebc0f4dd049))
* make the tumor cell content optional ([#160](https://www.github.com/hydra-genetics/cnv_sv/issues/160)) ([1c960c6](https://www.github.com/hydra-genetics/cnv_sv/commit/1c960c638ffdefea8b23296f8a49fbe65b82b557))


### Bug Fixes

* add missing space in command ([21e2767](https://www.github.com/hydra-genetics/cnv_sv/commit/21e2767e8741a6d06cb2b4fedc5928d21025f49c))
* add quite to picard for stdout ([8f09906](https://www.github.com/hydra-genetics/cnv_sv/commit/8f09906109194e6444e84c7b3fc6a41ecfebab73))
* added RTD for all reviewer rules and fix rule name ([3a81357](https://www.github.com/hydra-genetics/cnv_sv/commit/3a81357f9ce3d063d9a77403e06d1ea6a557b26a))
* allow missing TC when generating GATK VCF ([#162](https://www.github.com/hydra-genetics/cnv_sv/issues/162)) ([aa6fece](https://www.github.com/hydra-genetics/cnv_sv/commit/aa6fece935ea8587910b273da7cbf6b8674bfb68))
* bump snakemake version to support apptainer ([#151](https://www.github.com/hydra-genetics/cnv_sv/issues/151)) ([eb4b05d](https://www.github.com/hydra-genetics/cnv_sv/commit/eb4b05df754c2b6be16d5feed3244d00d9f4b5f1))
* **cnvpytor:** add bam index as input ([60c840a](https://www.github.com/hydra-genetics/cnv_sv/commit/60c840ad7c71dbfd66c0925becc210ae31dfa41a))
* handle that cns files can have different number of columns ([57da77b](https://www.github.com/hydra-genetics/cnv_sv/commit/57da77b993ea25be88116e75712b09411e45677e))
* incorrect type for copy number thresholds ([#164](https://www.github.com/hydra-genetics/cnv_sv/issues/164)) ([b549266](https://www.github.com/hydra-genetics/cnv_sv/commit/b549266994149bf7b4bdcc385456229b4f0eee77))
* pin pulp version to <2.8.0 ([1c960c6](https://www.github.com/hydra-genetics/cnv_sv/commit/1c960c638ffdefea8b23296f8a49fbe65b82b557))
* rm extra } in benchmark name ([e2f34ce](https://www.github.com/hydra-genetics/cnv_sv/commit/e2f34ce8d24515d7d1f980994f3d72e12878d4da))
* specify model when calling and filtering in cnvpytor, and threads in readdpth rule ([4347806](https://www.github.com/hydra-genetics/cnv_sv/commit/434780660b693fc7b9dbdf709cc4700be33de63c))
* **svdb:** use the overlap and extra params ([fe656a9](https://www.github.com/hydra-genetics/cnv_sv/commit/fe656a91dd0a74c566ef0b8bf3512609a071ab94))
* update container for rule pindel_update_vcf ([#161](https://www.github.com/hydra-genetics/cnv_sv/issues/161)) ([512a924](https://www.github.com/hydra-genetics/cnv_sv/commit/512a924beba54282757745c8c66475e78f52cca3))


### Performance Improvements

* add bcftools reheader to pindel vcf for correct samplename ([fcadaba](https://www.github.com/hydra-genetics/cnv_sv/commit/fcadaba56976bad8a6f2975558c5368c25016fb8))


### Documentation

* added RTD files and the rules for automap and cnvkit ([ce2f2ca](https://www.github.com/hydra-genetics/cnv_sv/commit/ce2f2ca0b4cdd6876968f638d3d30226f62bfe19))
* added RTD for all manta rules ([2571744](https://www.github.com/hydra-genetics/cnv_sv/commit/25717444f3a6a48825e95d2c4e756ec2af62c244))
* added RTD for all pindel rules ([b0ca4da](https://www.github.com/hydra-genetics/cnv_sv/commit/b0ca4dadbbada7bbad66c1274c125f58ef69447d))
* added RTD for all purecn rules ([7211b29](https://www.github.com/hydra-genetics/cnv_sv/commit/7211b298c138c8aaa693bca072b78885379d8532))
* added RTD for all smncaller rules ([ed55b6b](https://www.github.com/hydra-genetics/cnv_sv/commit/ed55b6b121c08fb4b4935fe231f493e563f41104))
* added RTD for all svdb rules ([1cdd6ba](https://www.github.com/hydra-genetics/cnv_sv/commit/1cdd6ba37ca9cfd3c21458188d3d04b706334112))
* added RTD for cnvpytor and exomedepth ([8fd490a](https://www.github.com/hydra-genetics/cnv_sv/commit/8fd490a8b73efc412797859678569ecba07ac783))
* added RTD for expansionhunter ([4f290ca](https://www.github.com/hydra-genetics/cnv_sv/commit/4f290ca1e0b69dc7540411d877d380644da69b7e))
* added RTD for gatk cnv rules ([a0a575e](https://www.github.com/hydra-genetics/cnv_sv/commit/a0a575ef4a37ac0014b67da8690f8241eeef68f9))
* added RTD for tiddit rules ([fea5964](https://www.github.com/hydra-genetics/cnv_sv/commit/fea5964ec64f5e184c3df75a43241485fbdfa8b0))
* added RTD for upd rules ([2eccfe5](https://www.github.com/hydra-genetics/cnv_sv/commit/2eccfe5bc01c485afc3996cd85cf0c3700a06175))
* dag graph ([7a654c2](https://www.github.com/hydra-genetics/cnv_sv/commit/7a654c2ddedecc9a47321f8120b955162fe66674))
* organise rules based on software ([36568d9](https://www.github.com/hydra-genetics/cnv_sv/commit/36568d9cc61789b3ee5e07af1d695c01d783e496))
* Update docs/softwares.md ([d3e3f67](https://www.github.com/hydra-genetics/cnv_sv/commit/d3e3f67f1e91e15223cc655133c944128c9e5a54))
* Update docs/softwares.md ([188ef59](https://www.github.com/hydra-genetics/cnv_sv/commit/188ef594302900cc0b37580e616775412fe79764))
* Update docs/softwares.md ([90ad983](https://www.github.com/hydra-genetics/cnv_sv/commit/90ad983f53a88117f0532bb8042c74c279d1d7d5))
* Update docs/softwares.md ([dd02db5](https://www.github.com/hydra-genetics/cnv_sv/commit/dd02db57a7d41c253c1e1d5e79b521d43a6f003d))
* Update docs/softwares.md ([6f16c13](https://www.github.com/hydra-genetics/cnv_sv/commit/6f16c13479c785cfa52f7c318a9c85aa6a2f5727))
* Update docs/softwares.md ([25b3195](https://www.github.com/hydra-genetics/cnv_sv/commit/25b31959177b1cd798e4896e47c5bab36c94bad3))
* Update docs/softwares.md ([c02990c](https://www.github.com/hydra-genetics/cnv_sv/commit/c02990c978a76a7d26ed8f1ce4f962e42bc566e3))
* update to new rule plugin ([aedbdfd](https://www.github.com/hydra-genetics/cnv_sv/commit/aedbdfd8d9f3084b4e49a0d7e9bb709da707fb10))
* Update workflow/schemas/resources.schema.yaml ([6136f71](https://www.github.com/hydra-genetics/cnv_sv/commit/6136f71519cb149297621d16cb16c31bf8017838))

### [0.4.1](https://www.github.com/hydra-genetics/cnv_sv/compare/v0.4.0...v0.4.1) (2023-05-04)


### Bug Fixes

* change to config.get in exomedepth_call ([34f5753](https://www.github.com/hydra-genetics/cnv_sv/commit/34f5753530450e6ae8ba51bd19c5bb0946a45d14))

## [0.4.0](https://www.github.com/hydra-genetics/cnv_sv/compare/v0.3.1...v0.4.0) (2023-05-02)


### Features

* add upd ([cc5a660](https://www.github.com/hydra-genetics/cnv_sv/commit/cc5a660a61d0f6d9b4a8a4bc6559cadfcf26f741))
* **exomedepth:** add checkpoint rule to allow reference file as input ([a75d1e5](https://www.github.com/hydra-genetics/cnv_sv/commit/a75d1e5939e994ba44e665d8f6ae537228e71b5c))
* **exomedepth:** Add option to run exomedepth with hg38 ([179bab3](https://www.github.com/hydra-genetics/cnv_sv/commit/179bab3da18e62a927b6b93d2daef7727ebdd0b6))
* remove conda support and testing ([0a3610d](https://www.github.com/hydra-genetics/cnv_sv/commit/0a3610d881bd6c5a95dc13754140220b6adf3c13))


### Bug Fixes

* automap input and integration test ([20ec062](https://www.github.com/hydra-genetics/cnv_sv/commit/20ec062be15dce5a3766f50254d3fdd73b86361e))
* **exomedepth:** chosse female ref when the peddy sex is NA ([4357702](https://www.github.com/hydra-genetics/cnv_sv/commit/4357702e36f4b472b57bf6643ea19f26bf6356b0))
* get output ([fb11ed2](https://www.github.com/hydra-genetics/cnv_sv/commit/fb11ed2415ae8e281adb6000b531eb7974ed1bea))
* get samples filename from config, close [#135](https://www.github.com/hydra-genetics/cnv_sv/issues/135) ([46b53c6](https://www.github.com/hydra-genetics/cnv_sv/commit/46b53c6f51a040ff12e2a3f6c74ff55a1e5379ff))
* linting ([e28d277](https://www.github.com/hydra-genetics/cnv_sv/commit/e28d27753c0373190050474ecee1b5637151f596))
* out_dir ([55a13f0](https://www.github.com/hydra-genetics/cnv_sv/commit/55a13f040a0ccf8a8ab24e02e3af3efa9e08bb82))
* output ([8b8af48](https://www.github.com/hydra-genetics/cnv_sv/commit/8b8af48e9e0b47224fb028eeff34850431ad7df9))
* tests for sutomap ([33a384e](https://www.github.com/hydra-genetics/cnv_sv/commit/33a384e5634d93a67cdb9b910ecfb632d40a38c1))
* tsv to pdf ([3aa56c7](https://www.github.com/hydra-genetics/cnv_sv/commit/3aa56c72aa6d2d626b08de057310cc264fec7235))
* typo ([d5cb0ca](https://www.github.com/hydra-genetics/cnv_sv/commit/d5cb0ca43f30e8b923a21dadc96c04b7c44b09f3))


### Documentation

* update compatibility ([0e1e381](https://www.github.com/hydra-genetics/cnv_sv/commit/0e1e381ac649bb48a0573f932053d48153e86beb))
* update compatibility ([fe31a4b](https://www.github.com/hydra-genetics/cnv_sv/commit/fe31a4b4b39cb5413c8875947d22156a416ea1e2))
* update disclaimer ([1d98fea](https://www.github.com/hydra-genetics/cnv_sv/commit/1d98feab3ab2f1c6be40909119c00baee9045a48))
* update disclaimer ([cbb7592](https://www.github.com/hydra-genetics/cnv_sv/commit/cbb7592ecf98ffb4245e7673bcbefde2a4759a82))
* update message format ([76eaa0b](https://www.github.com/hydra-genetics/cnv_sv/commit/76eaa0bca03ee6942ee83b63dd68f5bf256f364d))
* update message format ([4e4f73a](https://www.github.com/hydra-genetics/cnv_sv/commit/4e4f73ad6f3c3e2ba5cd0896475ee63e8ebe5533))
* updated output files ([d56c8d6](https://www.github.com/hydra-genetics/cnv_sv/commit/d56c8d6282382719dc2ce0c5751a2097f616f8b6))

### [0.3.1](https://www.github.com/hydra-genetics/cnv_sv/compare/v0.3.0...v0.3.1) (2023-01-31)


### Bug Fixes

* **purecn_copy_output:** add missing benchmark and params ([c10d758](https://www.github.com/hydra-genetics/cnv_sv/commit/c10d7587ac96403dbfbee6df4cd3436a08377df3))


### Documentation

* update compatibility file ([b9fd87d](https://www.github.com/hydra-genetics/cnv_sv/commit/b9fd87d767499226c34dedc11aebf73fa95d2152))

## [0.3.0](https://www.github.com/hydra-genetics/cnv_sv/compare/v0.2.0...v0.3.0) (2023-01-27)


### Features

* added default in function ([bf4f3f2](https://www.github.com/hydra-genetics/cnv_sv/commit/bf4f3f2399494742c9ed38ab24cd2fa619dde4d5))
* added function that extracts purity for purecn ([9cf298e](https://www.github.com/hydra-genetics/cnv_sv/commit/9cf298e8eda05dc02280a84aa440dafc5f59082f))
* added pathology option ([4b3adc5](https://www.github.com/hydra-genetics/cnv_sv/commit/4b3adc52718494c0673b1a8bf7252e9bcedc0295))
* added rule purecn_purity_file ([2bed973](https://www.github.com/hydra-genetics/cnv_sv/commit/2bed973dd8e9e2420d630978fe24571fec36a388))
* added tags for purity method ([b9f3d3d](https://www.github.com/hydra-genetics/cnv_sv/commit/b9f3d3d6b3a8ffbd39f32071d4eed12907703d6d))
* svdb is now configurable to tc method ([c3eb2e1](https://www.github.com/hydra-genetics/cnv_sv/commit/c3eb2e17e04eed00e8c0ea8314e59035560c4e4f))


### Bug Fixes

* added backslash in cmd ([994fa78](https://www.github.com/hydra-genetics/cnv_sv/commit/994fa783459192db0d6b039023a16a85907a4a13))
* allow purecn to run on all sample types ([6947acf](https://www.github.com/hydra-genetics/cnv_sv/commit/6947acf0a3e91aa3de0080cb9a907fe8af10b880))
* bugfx ([ff30237](https://www.github.com/hydra-genetics/cnv_sv/commit/ff30237dec88ebf163dffea7f8d55e930968c6e4))
* corrected get_tc function output ([3c89633](https://www.github.com/hydra-genetics/cnv_sv/commit/3c89633a26be3a315774b112640d467d5bb15705))
* input file now with germline annotation ([bb7e5dd](https://www.github.com/hydra-genetics/cnv_sv/commit/bb7e5ddb7c1bbb0305d4df546cd8d43002a8bc21))
* output files ([97fb714](https://www.github.com/hydra-genetics/cnv_sv/commit/97fb714803894e91600badb34b9a89b4483992b0))
* rename caller from gatk_cnv to gatk ([6c0d00f](https://www.github.com/hydra-genetics/cnv_sv/commit/6c0d00f149419c883935b80faa30d17ca483ecce))
* rule name and output folder ([003f415](https://www.github.com/hydra-genetics/cnv_sv/commit/003f4159f3a3a681ef393a94cde2ec1b5012784c))
* tc files only in params and does not trigger purecn ([28bdbe4](https://www.github.com/hydra-genetics/cnv_sv/commit/28bdbe4e99c6569b20cd33eaf71ad6d9be00205d))


### Documentation

* update CODEOWNERS ([c20ccbc](https://www.github.com/hydra-genetics/cnv_sv/commit/c20ccbcdee442834e139f5e510f92425b7cdad68))
* update compatibility ([6bb06a5](https://www.github.com/hydra-genetics/cnv_sv/commit/6bb06a5624d813f0a67b06bc4642a007c2bc4ffa))

## [0.2.0](https://www.github.com/hydra-genetics/cnv_sv/compare/v0.1.0...v0.2.0) (2022-11-22)


### Features

* Initial PureCN implementation ([827ecd5](https://www.github.com/hydra-genetics/cnv_sv/commit/827ecd59ce1f38d50e75eae16f58dcd5fdc3e906))
* run expansionhunter with sex from peddy ([96248bf](https://www.github.com/hydra-genetics/cnv_sv/commit/96248bfb2dd5f6625905d53de20c87ed4be9955d))
* update pindel call wrapper ([deca8ee](https://www.github.com/hydra-genetics/cnv_sv/commit/deca8eedbf4effddaafe51964d9760f9245e18e2))


### Bug Fixes

* Add cnvkit_seg output ([6b2bc02](https://www.github.com/hydra-genetics/cnv_sv/commit/6b2bc024ebd445f0a44ae10c9e09b8f0b5c4682c))
* Add dummy intervals file for dry-run ([c36efa8](https://www.github.com/hydra-genetics/cnv_sv/commit/c36efa8e753df8b3875ad766b69099d400e41ca8))
* Add mutect2 VCFs for PureCN ([58c4784](https://www.github.com/hydra-genetics/cnv_sv/commit/58c4784727c27977ffa6683a198197da35caa68a))
* Add required parameters for purecn ([a6771a6](https://www.github.com/hydra-genetics/cnv_sv/commit/a6771a635b97209d17356e59e65d7d9182c56c82))
* fix integration test problem ([139c948](https://www.github.com/hydra-genetics/cnv_sv/commit/139c948a7a3ab963abcd6b4d83f914b35ba22fc6))
* fun_segmentation is a required property for purecn ([2376c99](https://www.github.com/hydra-genetics/cnv_sv/commit/2376c995fccc36dd3bae807ad12c098d60cd0877))
* make run step of manta require bam as input to prevent bam files from being cleaned away ([5918c73](https://www.github.com/hydra-genetics/cnv_sv/commit/5918c739006e1b7e4b25b11428ffd5e90801bad9))
* Minor language tweaks ([aebacab](https://www.github.com/hydra-genetics/cnv_sv/commit/aebacabd1f309d4d39699c298144e6dc28f3f3bd))
* Re-order resources in rules ([1c889d4](https://www.github.com/hydra-genetics/cnv_sv/commit/1c889d4bdbd71df099d8af5485ddbff8e74406e9))
* Rename cnvkit_seg log and benchmark ([c9f2808](https://www.github.com/hydra-genetics/cnv_sv/commit/c9f2808ee711a00a95ad766481e334d6cb3b2e13))
* Resolve snakefmt warnings ([fa7651e](https://www.github.com/hydra-genetics/cnv_sv/commit/fa7651e8b4d37db4b9e4d38ffea0cc1c75094773))
* Resolve snakefmt warnings ([b738559](https://www.github.com/hydra-genetics/cnv_sv/commit/b738559ec662e794f834b731b581afc633f41ea3))
* Separate PureCN output rule ([ef80f60](https://www.github.com/hydra-genetics/cnv_sv/commit/ef80f603a6d5068ad3ebb5c9fde82df7edc38b0e))
* Simplify expected PureCN output ([22fb008](https://www.github.com/hydra-genetics/cnv_sv/commit/22fb0089163a29416462ec60c2e3d7dce9dcbcb8))
* **tiddit:** fix env and bam to work with tiddit ([6803a15](https://www.github.com/hydra-genetics/cnv_sv/commit/6803a150934b149925919ce1f6bfdc19f49c83d6))
* Typos in output definitions ([106edaa](https://www.github.com/hydra-genetics/cnv_sv/commit/106edaabd90bd589140743aed519c175f4eff89c))
* Use dummy intervals ([8bcdae0](https://www.github.com/hydra-genetics/cnv_sv/commit/8bcdae0300b7ff849b5bd5015939d2c2595bf2dc))
* Use hydragenetics container for cnvkit_seg ([65bd30b](https://www.github.com/hydra-genetics/cnv_sv/commit/65bd30b030d8774190e488fdaaad5b37ebf17985))
* Use same conda env for all PureCN rules ([a795870](https://www.github.com/hydra-genetics/cnv_sv/commit/a795870b2c0628e9baeb2e331d13540376e43f36))


### Documentation

* **README:** change header to regular text ([d0e74d4](https://www.github.com/hydra-genetics/cnv_sv/commit/d0e74d4fa7dad72523c41d3a8209d40aceeaf0bf))

## 0.1.0 (2022-10-14)


### Features

* Add 3 rules for pindel plus envs and script ([5c51855](https://www.github.com/hydra-genetics/cnv_sv/commit/5c5185593b7718fa49270f017bd26e33f63b1243))
* add conventional-prs workflow ([94cc587](https://www.github.com/hydra-genetics/cnv_sv/commit/94cc5875c86cd2cdce2dc43a2a2cf4792337811c))
* add exomedepth ([ebfe037](https://www.github.com/hydra-genetics/cnv_sv/commit/ebfe037a87564692ee89be8e7114ddc959a6a4d3))
* Add logging and make script testable ([167a148](https://www.github.com/hydra-genetics/cnv_sv/commit/167a148bcf91823bbf0cde5f720bfb0c54c8fb7c))
* add missing entry in schema ([61a7055](https://www.github.com/hydra-genetics/cnv_sv/commit/61a70557f910f60ef3b9d7a4802f194c33f62cbb))
* Add rule cnvkit_call_loh which adapt calls according to TC ([3837d51](https://www.github.com/hydra-genetics/cnv_sv/commit/3837d51dfd94e11f4617fb7c5e887ea595376998))
* Add rule to generate cnvkit diagram ([bef8828](https://www.github.com/hydra-genetics/cnv_sv/commit/bef8828521497758e3c9f685dda009c6ca10a721))
* added bedfile to pindel call ([36aca30](https://www.github.com/hydra-genetics/cnv_sv/commit/36aca300ac068610384c0534bfe618b64d66aa5b))
* Added environments ([c919fed](https://www.github.com/hydra-genetics/cnv_sv/commit/c919fedec6e2b1601c8740fdbae8b178907a57ac))
* Added manta rules ([e77071d](https://www.github.com/hydra-genetics/cnv_sv/commit/e77071d8a702e85af9fcc73025bdc7bae29c7ba0))
* Added rule cnvkit coverage ([7162e00](https://www.github.com/hydra-genetics/cnv_sv/commit/7162e0084526b6782f986087f7ca5d735857a062))
* Added rule cnvkit_call together with test files ([4d7c808](https://www.github.com/hydra-genetics/cnv_sv/commit/4d7c80838866c49cd9dd33ffe124cd09df57e334))
* Added rule cnvkit_create_access ([7d5a550](https://www.github.com/hydra-genetics/cnv_sv/commit/7d5a550606a2f8134bf347e7b7ebca31a35bd887))
* Added rule cnvkit_create_antitargets ([5fd5d12](https://www.github.com/hydra-genetics/cnv_sv/commit/5fd5d12ea96a4f220823b93e233fa0ed9904ca4a))
* Added rule cnvkit_create_targets ([cdb3f5e](https://www.github.com/hydra-genetics/cnv_sv/commit/cdb3f5e14cd0de28173e48a2ebff8ff2a0ef3d8b))
* Added rule for manta tumor only calling. Refactor of old rule ([530a295](https://www.github.com/hydra-genetics/cnv_sv/commit/530a29503eecabc46fc814c7eb031fc0db6bb467))
* Added rule GATK_cnv_collectAllelicCounts ([d4abf5e](https://www.github.com/hydra-genetics/cnv_sv/commit/d4abf5ed996794397ee24cd3abd0640549e54d3b))
* Added rule GATK_cnv_collectReadCounts ([66ab8f7](https://www.github.com/hydra-genetics/cnv_sv/commit/66ab8f7e1163723f0fee38bab1c7ccc8d1befda4))
* Added rule GATK_cnv_denoiseReadCounts ([99822b5](https://www.github.com/hydra-genetics/cnv_sv/commit/99822b57882a5b17bdadae43d31445f279598371))
* Added rule GATK_cnv_modelSegments ([8c88d06](https://www.github.com/hydra-genetics/cnv_sv/commit/8c88d06e8d210117f2e38e6f7759d94332a1a9fc))
* Added rule germline_vcf ([9bbabc4](https://www.github.com/hydra-genetics/cnv_sv/commit/9bbabc42345a7cf416c2fc292287c213987611b8))
* Changed cnvkit_vcf to own script ([db9a85d](https://www.github.com/hydra-genetics/cnv_sv/commit/db9a85d03ba086ac4650f163d935e9640369abb8))
* Generate output list programmatically ([ef95803](https://www.github.com/hydra-genetics/cnv_sv/commit/ef9580374e6b9b72549183b37f8d3742f8b68e94))
* germline vcf now performed by filter vcf using pipeline config and override in Snakemake ([0a427b6](https://www.github.com/hydra-genetics/cnv_sv/commit/0a427b605f4ce1eea3fa8d8b0f1ff4dfcf3b8b80))
* make config.yaml location more flexible ([f6f4416](https://www.github.com/hydra-genetics/cnv_sv/commit/f6f4416ec0594aaa92e681266fa33085a095127d))
* make configfile/confgilefiles argument mandatory ([02b4f25](https://www.github.com/hydra-genetics/cnv_sv/commit/02b4f25baa9dfd27cfcbbc19a6b6fe293c1962ad))
* make module compatible with latest version of snv_indels and alignment module. ([8ea6383](https://www.github.com/hydra-genetics/cnv_sv/commit/8ea638395a2fbed2d8814747b748ca4766bbf4e6))
* make output more configurable ([dbd9bbf](https://www.github.com/hydra-genetics/cnv_sv/commit/dbd9bbf3b32aaa2f22f265dc5df22fffee5e214a))
* Make unnecessary manta output vcfs temp ([46688c2](https://www.github.com/hydra-genetics/cnv_sv/commit/46688c2a213415323142d439a7908acc3fa67390))
* **manta:** added extra params to manta ([e99460f](https://www.github.com/hydra-genetics/cnv_sv/commit/e99460f224c86057478ffb1f8fcdda30b5577926))
* new rule cnvkit_vcf that exports segment files to vcf ([9769357](https://www.github.com/hydra-genetics/cnv_sv/commit/976935749710ddcadb0da0385f02351716f98eac))
* New rule svdb ([e7349b4](https://www.github.com/hydra-genetics/cnv_sv/commit/e7349b4417a833876576a32a1fc1030257e9487e))
* Prepare pindel config script for unit testing ([13b0264](https://www.github.com/hydra-genetics/cnv_sv/commit/13b02646689787c6c34151e3822a0ae8749cbc8f))
* setup integration test for exomedepth ([7a30fc0](https://www.github.com/hydra-genetics/cnv_sv/commit/7a30fc0697bc401aa5e63b5f12e733d01a835d77))
* update snakemake-version ([830124c](https://www.github.com/hydra-genetics/cnv_sv/commit/830124cb81318ec9eb631806b4e82b5e0ea5af4b))
* update svdb version ([d6745f6](https://www.github.com/hydra-genetics/cnv_sv/commit/d6745f65ae7b84182af5a5f5e4aeca12f4590392))
* Upgrade pindel to wrapper v85.0.1 ([9048434](https://www.github.com/hydra-genetics/cnv_sv/commit/9048434f3c08f9db153e26e67702d3c63bacb2d5))


### Bug Fixes

* add containers and change from bcftools view to zcat in rule germinline_vcf ([5fa941f](https://www.github.com/hydra-genetics/cnv_sv/commit/5fa941f7aa8f34492a712bece844e725b727d177))
* add contigs to pindel-vcf ([4a52b00](https://www.github.com/hydra-genetics/cnv_sv/commit/4a52b0038e18c2301e80106eec4017e5b4a6cf3f))
* add library to install for exomedepth ([63d19c2](https://www.github.com/hydra-genetics/cnv_sv/commit/63d19c221ff679f7dda47976dee62a965ca2fdec))
* Added resources, rule name fix, rm testfiles, rm wp1 in config ([f408a14](https://www.github.com/hydra-genetics/cnv_sv/commit/f408a14af5893ae2e034a29cfb3d7b75a62fa448))
* better vcf annotation for filtering ([8aa6da8](https://www.github.com/hydra-genetics/cnv_sv/commit/8aa6da8bed5b35d88c86cb45809c2ef92996d445))
* bugfix ([140842c](https://www.github.com/hydra-genetics/cnv_sv/commit/140842c587a64e7b25183c0b58ad67b903a850b0))
* change run column name to flowcell in units. ([a69af37](https://www.github.com/hydra-genetics/cnv_sv/commit/a69af37ad1759e8e155527655d056e399494e7b6))
* change TC to tumor_content ([8c11ee6](https://www.github.com/hydra-genetics/cnv_sv/commit/8c11ee695e48331bd9a829b4e6808129eea67457))
* change to point to master ([41e6d16](https://www.github.com/hydra-genetics/cnv_sv/commit/41e6d166dddcd5d0bccdbaabb7b42b81ca981079))
* change type from float to number ([1731573](https://www.github.com/hydra-genetics/cnv_sv/commit/1731573fb98df66c6eb497bbc5851a1b82632521))
* Changed name again ([ea530f1](https://www.github.com/hydra-genetics/cnv_sv/commit/ea530f1c59da79dc95b71721d907e5247ea33c23))
* **cnvkit:** change python version ([f315187](https://www.github.com/hydra-genetics/cnv_sv/commit/f31518771ec929884e222977b5d7e31a50e72374))
* **cnvkit:** change to correct conda env file ([be3d3ba](https://www.github.com/hydra-genetics/cnv_sv/commit/be3d3ba5d3a9bca28daa23031f93684723182608))
* Correct runWorkflow script path and add container to config ([057b151](https://www.github.com/hydra-genetics/cnv_sv/commit/057b1511bf66fe80acb0dc0ceba34233c208477c))
* don't install snakemake using mamba ([dc08143](https://www.github.com/hydra-genetics/cnv_sv/commit/dc08143c1c088d91d8b09a46c92029e846fd56fe))
* float in header ([4e30d39](https://www.github.com/hydra-genetics/cnv_sv/commit/4e30d39b5470061211cbc1577722a797381a0f03))
* float in header ([7c0c233](https://www.github.com/hydra-genetics/cnv_sv/commit/7c0c23333eed9a0f1d5779d150445c8add3c9d7d))
* handling of multiple databases ([978d218](https://www.github.com/hydra-genetics/cnv_sv/commit/978d2185aa2a0a0c9b282ee6099872577ceb9ba2))
* info ids to long ([541bd8d](https://www.github.com/hydra-genetics/cnv_sv/commit/541bd8d4e8df4566872e82ed7b475482900370d0))
* int to float in vcf headers ([331da1d](https://www.github.com/hydra-genetics/cnv_sv/commit/331da1d554ecddec9c5e5be2f328a097020a9308))
* lock singularity version to prevent bug with latest version ([bfe978e](https://www.github.com/hydra-genetics/cnv_sv/commit/bfe978ea5e29d00fd6835e23b090f5eb5020d877))
* reference under rule ([92fd035](https://www.github.com/hydra-genetics/cnv_sv/commit/92fd03538d20ae7043606fd3f8594f77f3ddcec4))
* remove unused dependency. ([81f7ed6](https://www.github.com/hydra-genetics/cnv_sv/commit/81f7ed66695ed3a472bad33c6b5646dee25eb27c))
* revert changes ([e74b497](https://www.github.com/hydra-genetics/cnv_sv/commit/e74b497594df70e11e881f95c169d958fc740432))
* rule name change ([f617b06](https://www.github.com/hydra-genetics/cnv_sv/commit/f617b0655fe332b359407f7c3ed357a68e222d5c))
* set correct env file ([c81d30d](https://www.github.com/hydra-genetics/cnv_sv/commit/c81d30de64d4cc291e32b270c3a8a763c10d7fa5))
* set strict mode for conda ([d070bda](https://www.github.com/hydra-genetics/cnv_sv/commit/d070bdab490f726339e5a03263e674dd67b58612))
* update r script to handle empty results ([6bd463b](https://www.github.com/hydra-genetics/cnv_sv/commit/6bd463b141a14ba16669f59b816ad49b7580824f))
* used corrected cn instead of cn ([dc59207](https://www.github.com/hydra-genetics/cnv_sv/commit/dc5920740d9cc54f12e73ca5390fb096f2d2f2a8))


### Documentation

* add and fix schema for picard_update_vcf_sequence_dictionary ([9bba1e9](https://www.github.com/hydra-genetics/cnv_sv/commit/9bba1e9cb72bf3b0c3e966006e61372b3bb6971a))
* add configfile option to example ([da7ff6b](https://www.github.com/hydra-genetics/cnv_sv/commit/da7ff6b239181bf9ac13cae94a68c7fe93045220))
* add description to all config- and resources-schemas ([d7103d4](https://www.github.com/hydra-genetics/cnv_sv/commit/d7103d40fbfa5c102dfcc7b27fcba50e4c29e2eb))
